### ISSUE TITLE
Removed vanilla crafting recipes still showing up in JEI

### ISSUE THREAD

**Issue by al132mc** [codetaylor/pyrotech#97]
#### Issue Description

I have many vanilla crafting recipes disabled in ``.core.cfg`` (not the default list, but I left most of them). They seem to be disabled just fine, but they still show up in JEI

#### What Happens

See above

#### What You Expect to Happen

Ideally these removed vanilla crafting recipes would not show in JEI

#### Script

[.core.cfg](https://pastebin.com/rBg9i5Sf)

#### Affected Versions

* Minecraft: 1.12.2
* Forge: 14.23.5.2838
* CraftTweaker: 1.12-4.1.19
* Dropt: 1.12.2-1.15.1
* Pyrotech: 1.12.2-1.2.1
* Athenaeum: 1.12.2-1.17.2

![2019-07-03_12 38 41](https://user-images.githubusercontent.com/39097092/60613354-56e1c980-9d90-11e9-8577-982de9f282a5.png)
![2019-07-03_12 38 43](https://user-images.githubusercontent.com/39097092/60613355-56e1c980-9d90-11e9-8e6f-d543f0cae611.png)

**Comment by codetaylor** [comment]
Which version of JEI are you using?

**Comment by al132mc** [comment]
``jei_1.12.2-4.15.0.285.jar	``
which is the most recent currently

**Comment by codetaylor** [comment]
I have confirmed that this is happening with the versions indicated.

![image](https://user-images.githubusercontent.com/1240716/60616623-c5ba2500-9d86-11e9-8d51-ec8d58b4d02e.png)


![image](https://user-images.githubusercontent.com/1240716/60616639-cfdc2380-9d86-11e9-8dc8-7f9066d73574.png)

**Comment by codetaylor** [comment]
It looks like this is happening because in `1.2.0` the recipe removal was moved to the `LoadComplete` event. This is also the event [during which JEI starts all its plugins](https://github.com/mezz/JustEnoughItems/blob/1.12/src/main/java/mezz/jei/startup/ProxyCommonClient.java#L118), so this is probably the conflict. I'm not entirely confident in how the FML lifecycle events are ordered. I'm using an older version of JEI in my dev environment, and I haven't experienced this, which would indicate that the `LoadComplete` event was handled first by Pyrotech.

The evidence suggests that with the newer versions:
1. JEI initializes its plugins, and then
2. Pyrotech removes recipes

I'm going to look into maybe having Pyrotech check if JEI has started before removing the recipes. If it has started, hide the recipes in JEI.

**Comment by codetaylor** [comment]
Confirmed this approach seems to work and the behavior is consistent across versions.

JEI is still gathering the removed recipes for the worktable though, so those must be removed too.

**Comment by codetaylor** [comment]
The removed smelting recipes are also suffering from the same problem, they will also have to be hidden.

**Comment by codetaylor** [comment]
This approach has hit a snag. It works fine for crafting recipes because those all have a unique identifier, however, smelting recipes do not have a unique identifier. Smelting recipes are identified by their output and, since the removal code is run in the `LoadComplete` event, it is run after CrT; any smelting recipes added by CrT that matched the output listed in Pyrotech's config would then be hidden in JEI.

I'm trying to think of a case where this would be an issue.

Ok, this is actually an issue in Pyrotech right now:

![image](https://user-images.githubusercontent.com/1240716/60621800-87773280-9d93-11e9-862a-2d4d67889647.png)

![image](https://user-images.githubusercontent.com/1240716/60621830-99f16c00-9d93-11e9-9e34-f79f6a56439b.png)

JEI displays the removed recipe as well as the new recipe. However, since the output of the new recipe added by Pyrotech matches an output to be removed, it gets removed as well...

Recipe removal was moved to the `LoadComplete` event in order to prevent Pyrotech from removing recipes before Thermal Expansion had a chance to generate its recipes from the removed stuff. Specifically, the log to planks crafting recipes. I don't know if Pyrotech's removal of furnace recipes interfered with Thermal's recipe gathering, but I moved it to the `LoadComplete` event anyway.

This has been a lot of work so far and it looks like it will be causing even more headaches.

If I move the removal back to where it was before. Thermal Expansion will be missing a few common recipes, such as the sawmill recipes for turning logs into planks. Other mods that rely on state interrogation to derive recipes might also suffer for this.

In the case where another mod suffers recipe loss due to Pyrotech, the user could be instructed to remove the recipes of interest from Pyrotech's config. This would re-enable the crafting recipes that Pyrotech removes by default and at that point it would be up to the user to remove them using other means.

If the user were to stop Pyrotech from removing the recipes and then remove them with CrT, for example, Thermal Expansion might still be missing the recipes. I don't know when TE gathers its recipe data. At least then it would no longer be the fault of Pyrotech.
